src/en/developer-resources.md

```diff
@@ -0,0 +1,131 @@
+Title: Writing charms using resources
+
+# Writing charms that use resources
+
+Many applications require binary resources in order to be installed. While it is
+possible for a charm to download the software from the package repositories, or
+other locations, some charms may be deployed network restricted environments
+that do not allow access to all areas of the Internet.
+
+Starting with version 2 of Juju, users can upload resources to the controller
+that charms can download. This is useful for Juju models in restrictive
+network environments and when you want to control the versions of software that
+is deployed.
+
+# How it works
+
+## Developing a charm with resources
+
+Charm developers can add a `resources` key to the  `metadata.yaml` file to
+define one or more resources.
+
+```yaml
+resources:
+  software:
+    type: file
+    filename: software.zip
+    description: "One line description that is useful when operators need to push it."
+```
+The `filename` is the name of the resource after it has been retrieved. Juju
```

Comment:
"The filename field contains the name of the file that will be created on the unit after the resource has been retrieved from the juju controller."